modal/_runtime/user_code_imports.py

```diff
@@ -226,11 +226,15 @@ def import_single_function_service(
     """
     user_defined_callable: Callable
     service_deps: Optional[Sequence["modal._object._Object"]] = None
-    active_app: Optional[modal.app._App] = None
+    active_app: modal.app._App
+
+    user_cls_or_cls: typing.Union[None, type, modal.cls.Cls]
```

Comment:
Why Union here rather than Optional? (jw if there's a reason)

Reply:
(I still prefer to use Optional[] when the arg type is a non-union, if that makes sense